pkg/kubelet/kubelet.go

```diff
@@ -1271,6 +1272,9 @@ func (kl *Kubelet) initializeModules() error {
 		}
 	}
 
+	// Adding Registration Callback function for CSI Driver
+	kl.pluginWatcher.AddHandler("CSIPlugin", csi.RegistrationCallback)
```

Comment:
Move this to inside `NewVolumeManager(...)`. And modify `NewVolumeManager(...)` to take `pluginWatcher pluginwatcher.Watcher` as a new parameter. That way all volume plugin code is inside `VolumeManager`